src/compo/viirs_aod2ioda.py

```diff
@@ -117,65 +260,153 @@ def _read(self):
                 qcall = qcall[mask]
                 obs_time = obs_time[mask]
 
-            ncd.close()
-
-            # apply thinning mask
+                if self.adp_mask is not None:
+
+                    mask_smoke = np.logical_not(vals_smoke.mask)
+                    vals_smoke = vals_smoke[mask_smoke]
+                    lons_smoke = lons_smoke[mask_smoke]
+                    lats_smoke = lats_smoke[mask_smoke]
+                    errs_smoke = errs_smoke[mask_smoke]
+                    qcpath_smoke = qcpath_smoke[mask_smoke]
+                    qcall_smoke = qcall_smoke[mask_smoke]
+                    obs_time_smoke = obs_time_smoke[mask_smoke]
+
+                    mask_dust = np.logical_not(vals_dust.mask)
+                    vals_dust = vals_dust[mask_dust]
+                    lons_dust = lons_dust[mask_dust]
+                    lats_dust = lats_dust[mask_dust]
+                    errs_dust = errs_dust[mask_dust]
+                    qcpath_dust = qcpath_dust[mask_dust]
+                    qcall_dust = qcall_dust[mask_dust]
+                    obs_time_dust = obs_time_dust[mask_dust]
+
+            # Apply thinning mask
             if self.thin > 0.0:
+
                 mask_thin = np.random.uniform(size=len(lons)) > self.thin
+                vals = vals[mask_thin]
                 lons = lons[mask_thin]
                 lats = lats[mask_thin]
-                vals = vals[mask_thin]
                 errs = errs[mask_thin]
                 qcpath = qcpath[mask_thin]
                 qcall = qcall[mask_thin]
                 obs_time = obs_time[mask_thin]
 
+                if self.adp_mask is not None:
+
+                    # switch smoke and dust to just use mask_thin if moved above maskout block
+                    mask_thin_smoke = np.random.uniform(size=len(lons_smoke)) > self.thin
+                    vals_smoke = vals_smoke[mask_thin_smoke]
+                    lons_smoke = lons_smoke[mask_thin_smoke]
+                    lats_smoke = lats_smoke[mask_thin_smoke]
+                    errs_smoke = errs_smoke[mask_thin_smoke]
+                    qcpath_smoke = qcpath_smoke[mask_thin_smoke]
+                    qcall_smoke = qcall_smoke[mask_thin_smoke]
+                    obs_time_smoke = obs_time_smoke[mask_thin_smoke]
+
+                    mask_thin_dust = np.random.uniform(size=len(lons_dust)) > self.thin
+                    vals_dust = vals_dust[mask_thin_dust]
+                    lons_dust = lons_dust[mask_thin_dust]
+                    lats_dust = lats_dust[mask_thin_dust]
+                    errs_dust = errs_dust[mask_thin_dust]
+                    qcpath_dust = qcpath_dust[mask_thin_dust]
+                    qcall_dust = qcall_dust[mask_thin_dust]
+                    obs_time_dust = obs_time_dust[mask_thin_dust]
+
+            # calculate and store ob errors
             # defined surface type and uncertainty
             if self.method == "nesdis":
```

Comment:
This scripts looks only support the method of "nesdis". If it is not "nesdis", the variable "errs" become undefined. If so, please update the document mentioning that this script only support nesdis product. The zero observation error could cause some issues in assimilation.

Reply:
Good point @ytangnoaa. Based on the documentation, it looks like when that method selection conditional was added there was supposed to be an else block that set the errors to zero if the nesdis method wasn't specified with -m/--method. So the current documentation is incorrect regarding the default behavior. 

I could add in the missing else block to set the errors to zero, but given that both undefined and zero error values could cause problems during DA, I would propose changing the default behavior to use the nesdis method and then updating the documentation accordingly. That way the method option will remain available at the command line for existing scripts that call the code using that option, but the ability to use a method other than nesdis will be gone unless additional methods are added in later. If you think anyone may for some reason be using the converter with the expectation of it following the current default of leaving the error values undefined, though, then I will just update the documentation. 

For clarity, the options I see are:
1.) Leave the code as is, update the documentation to note that the default behavior leaves the error values undefined and is not recommended.
2.) Add in the else block to set the error values to zero for the default case to match the behavior stated in the current documentation, but also update the documentation to note that the default zero error values may cause problems.
3.) Change the default value for the -m/--method argument from none to nesdis and update the documentation to note that the nesdis method is the only method currently available, but the argument remains available for ease of possible future expansion (and for current backward compatibility).

Reply:
We should have a default option that do make sense and should avoid creating strictly 0 errors. This WILL cause problems in data assimilation. The code will crash.  So we should modify the code (the if statement in question here) to avoid creating zeros as errors and make sure that we use the nesdis option or whatever allowed options we could have in the future. 
Ok with solution 3. And update the the help argument to something that is not confusing: for example: `obs error calculation method available: nesdis, user must specify it or the converter won't work`.
Thanks